packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts

```diff
@@ -0,0 +1,485 @@
+              id,
+              nabuId,
+              origin: 'SIGN_IN',
+              originalTimestamp: getOriginalTimestamp(),
+              type: AnalyticsType.EVENT
+            })
+
+            analytics.push(AnalyticsKey.DASHBOARD_VIEWED, {
+              id,
+              nabuId,
+              originalTimestamp: getOriginalTimestamp(),
+              path: pathname,
+              referrer,
+              search,
+              title,
+              type: AnalyticsType.EVENT,
+              url: href
+            })
+
+            analytics.clear()
+
+            break
+          }
+          default: {
+            break
+          }
+        }
+
+        break
+      }
+      case AT.modals.SHOW_MODAL: {
+        const modalName: ModalNamesType = action.payload.type
+
+        switch (modalName) {
+          case 'SIMPLE_BUY_MODAL': {
+            const rawOrigin = action.payload.props.origin
+            const { href, pathname, search } = window.location
+            const { referrer, title } = document
+
+            const origin = simpleBuyOriginDictionary(rawOrigin)
+
+            analytics.push(AnalyticsKey.BUY_SELL_CLICKED, {
+              id,
+              nabuId,
+              origin,
+              originalTimestamp: getOriginalTimestamp(),
+              type: AnalyticsType.EVENT
+            })
+
+            analytics.push(AnalyticsKey.BUY_SELL_VIEWED, {
+              id,
+              nabuId,
+              originalTimestamp: getOriginalTimestamp(),
+              path: pathname,
+              referrer,
+              search,
+              title,
+              type: AnalyticsType.EVENT,
+              url: href
+            })
+
+            break
+          }
+          case 'SWAP_MODAL': {
+            const { origin } = action.payload.props
+            const { href, pathname, search } = window.location
+            const { referrer, title } = document
+
+            analytics.push(AnalyticsKey.SWAP_CLICKED, {
+              id,
+              nabuId,
+              origin,
+              originalTimestamp: getOriginalTimestamp(),
+              type: AnalyticsType.EVENT
+            })
+
+            analytics.push(AnalyticsKey.SWAP_VIEWED, {
+              id,
+              nabuId,
+              originalTimestamp: getOriginalTimestamp(),
+              path: pathname,
+              referrer,
+              search,
+              title,
+              type: AnalyticsType.EVENT,
+              url: href
+            })
+
+            break
+          }
+          default: {
+            break
+          }
+        }
+
+        break
+      }
+
+      case AT.components.interest.SET_COIN_DISPLAY: {
+        const { isCoinDisplayed } = action.payload
+
+        const fix = isCoinDisplayed ? 'CRYPTO' : 'FIAT'
+
+        analytics.push(AnalyticsKey.AMOUNT_SWITCHED, {
+          id,
+          nabuId,
+          originalTimestamp: getOriginalTimestamp(),
+          product: 'SAVINGS',
```

Comment:
can we start naming this `INTEREST`?

Reply:
It’s everywhere as `SAVINGS` in Nabu. So to make it more reliable on all the devices, it's `SAVINGS` here.